### PR TITLE
[CARBONDATA-687] Added documentation for new features in the DML, DDL Section and Troubleshooting

### PR DESCRIPTION
**Pull request by PallaviSingh1992**

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/789/

File: docs/dml-operation-on-carbondata.md [hunk 1]
```diff
@@ -133,6 +133,25 @@ You can use the following options to load data:
 
     NOTE: Date formats are specified by date pattern strings. The date pattern letters in CarbonData are same as in JAVA. Refer to [SimpleDateFormat](http://docs.oracle.com/javase/7/docs/api/java/text/SimpleDateFormat.html).
 
+- **USE_KETTLE:** This option is used to specify whether to use kettle for loading data or not. By default kettle is used for data loading.
```
Comment (jackylk) [inline docs/dml-operation-on-carbondata.md]: By default, kettle is not used

File: docs/dml-operation-on-carbondata.md [hunk 2]
```diff
@@ -133,6 +133,25 @@ You can use the following options to load data:
...
+
+    ```
+    OPTIONS('USE_KETTLE'='FALSE')
+    ```
+
+   Note :  It is recommended to set the value for this option as false.
+
+- **SINGLE_PASS:** This option specifies whether to use single pass for loading data or not. By default this option is set to FALSE.
```
Comment (jackylk) [inline docs/dml-operation-on-carbondata.md]: can you add description for what is single pass

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/798/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/799/

**Comment by sgururajshetty** [comment]
@chenliang613 please review and let us know any changes required.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/803/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/804/

File: docs/troubleshooting.md [hunk 3]
```diff
@@ -18,13 +18,249 @@
...
- * When trying to create a table with a single numeric column, table creation fails: 
-   One column that can be considered as dimension is mandatory for table creation.
-         
- * "Files locked for updation" when same table is accessed from two or more instances: 
-    Remove metastore_db from the examples folder.
+## Failed to load thrift libraries
 
+  **Symptom**
+
+  Thrift throws following exception :
```
Comment (chenliang613) [inline docs/troubleshooting.md]: For thrift configuration issues or installation problems, it is better to remind users to follow Apache thrift docs : https://thrift.apache.org/docs/install

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/868/

File: docs/ddl-operation-on-carbondata.md [hunk 4]
```diff
@@ -27,18 +27,19 @@ The following DDL operations are supported in CarbonData :
 * [SHOW TABLE](#show-table)
 * [DROP TABLE](#drop-table)
 * [COMPACTION](#compaction)
+* [BUCKETING](#bucketing)
 
 ## CREATE TABLE
   This command can be used to create a CarbonData table by specifying the list of fields along with the table properties.
-
```
Comment (chenliang613) [inline docs/ddl-operation-on-carbondata.md]: Why need to append one more rows at here ? please remove all needless format changes in this PR.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/890/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/891/

**Comment by chenliang613** [comment]
LGTM

**Pull request closed without merging**